omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/AlertResource2_0.java

```diff
@@ -0,0 +1,270 @@
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.notification.Alert;
+import org.openmrs.notification.AlertRecipient;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * {@link Resource} for {@link Alert}, supporting standard CRUD operations
+ */
+@Resource(name = RestConstants.VERSION_1 + "/alert", supportedClass = Alert.class, supportedOpenmrsVersions = { "2.0.*",
+		"2.1.*", "2.2.*", "2.3.*", "2.4.*", "2.5.*" })
+public class AlertResource2_0 extends DelegatingCrudResource<Alert> {
+
+	private static final String ALERT_ID = "alertId";
+
+	private static final String TEXT = "text";
+
+	private static final String DATE_TO_EXPIRE = "dateToExpire";
+
+	private static final String SATISFIED_BY_ANY = "satisfiedByAny";
+
+	private static final String ALERT_READ = "alertRead";
+
+	private static final String CREATOR = "creator";
+
+	private static final String DATE_CREATED = "dateCreated";
+
+	private static final String CHANGED_BY = "changedBy";
+
+	private static final String DATE_CHANGED = "dateChanged";
+
+	private static final String RECIPIENTS = "recipients";
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("display");
+		description.addProperty(ALERT_ID);
+		description.addSelfLink();
+
+		if (rep instanceof DefaultRepresentation) {
+			description.addProperty(TEXT);
+			description.addProperty(SATISFIED_BY_ANY);
+			description.addProperty(DATE_TO_EXPIRE);
+			description.addProperty(ALERT_READ);
+			description.addProperty(RECIPIENTS, Representation.REF);
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		} else if (rep instanceof FullRepresentation) {
+			description.addProperty(TEXT);
+			description.addProperty(SATISFIED_BY_ANY);
+			description.addProperty(ALERT_READ);
+			description.addProperty(DATE_TO_EXPIRE);
+			description.addProperty(CREATOR, Representation.DEFAULT);
+			description.addProperty(DATE_CREATED);
+			description.addProperty(CHANGED_BY, Representation.DEFAULT);
+			description.addProperty(DATE_CHANGED);
+			description.addProperty(RECIPIENTS, Representation.DEFAULT);
+		}
+
+		return description;
+	}
+
+	@PropertyGetter("display")
+	public String getDisplayProperty(Alert alert) {
+		return alert.getText();
+	}
+
+	@PropertySetter(RECIPIENTS)
+	public static void setRecipientsProperty(Alert alert, Set<AlertRecipient> newRecipients)
```

Comment:
If these are new recipients, shouldn't we just add them to the alert?

Reply:
These may be also updated alerts. For example, if previously we've had recipient ABC and now we're updating it with BCD then I will get hibernate error on save, that in this session there are more than 1 copies of BC objects.